topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -1234,17 +1234,20 @@ public void addKafkaTopics(List<KafkaTopic> ktList) {
                                 return Future.succeededFuture();
                             } else {
                                 LOGGER.debug("{}: Have private topic for topic {} in Kafka", logContext, topicName);
-                                Future<Void> map = reconcileWithPrivateTopic(logContext, topicName, topic, this)
+                                Promise<Void> map = Promise.promise();
+                                        reconcileWithPrivateTopic(logContext, topicName, topic, this)
                                         .<Void>map(ignored -> {
                                             LOGGER.debug("{} reconcile success -> succeeded", topicName);
                                             succeeded.add(topicName);
+                                            map.complete();
                                             return null;
                                         }).otherwise(error -> {
                                             LOGGER.debug("{} reconcile error -> failed", topicName);
                                             failed.put(topicName, error);
+                                            map.fail(error.getCause());
                                             return null;
                                         });
-                                return map;
+                                return map.future();
```

Comment:
I'm guessing that the old code didn't work because both the `map` and the `otherwise` were returning null, rather than a succeeding or failing `Future`. I think the code would be clearer if we fixed it that way.

Reply:
At this point, it means that we cannot use either `map` nor `otherwise` because they don't return a `Future` but the corresponding value which is of `Void` type in this case. We should change on using `setHandler` or `compose` (with success and failure lambas).